src/components/Dashboard.jsx

```diff
@@ -1,7 +1,217 @@
-import React from "react";
+import React, { useContext, useEffect, useState } from "react";
+import { UserContext } from "../context/UserContext";
+import { useRef } from "react";
 
 const Dashboard = () => {
-  return <div className="container">Dashboard</div>;
+  const [posts, setPosts] = useState([]);
+  // const [likedIds, setLikedIds] = useState([]);
+  const textareaRef = useRef();
+  const [post, setPost] = useState("");
+  const { name } = useContext(UserContext);
+
+  useEffect(() => {
+    let retrievedPosts;
+    if (localStorage.getItem("posts") !== null) {
+      retrievedPosts = JSON.parse(localStorage.getItem("posts"));
+      setPosts(retrievedPosts);
+    }
+  }, []);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const newPost = {
+      id: new Date().getTime(),
+      title: post,
+      date: new Date(),
+      author: name,
+      isPostOpen: false,
```

Comment:
Bhai what's this `isPostOpen` key does?

Reply:
It's for the `comment` box to toggle between open and close on click.